src/controllers/pools/post.controller.ts

```diff
@@ -25,7 +26,19 @@ const controller = async (req: Request, res: Response) => {
 
     const pool = await AssetPoolService.deploy(req.auth.sub, req.body.chainId, req.body.variant, req.body.tokens);
 
-    await pool.save();
+    const client = await ClientProxy.create(req.auth.sub, String(pool._id), {
```

Comment:
@GarfDev The tests in your PR were failing because you had removed client creation from this controller and some tests use the client created here to verify that access based on the client_credentials grant functions correctly. Two approaches here could have been:

1. Fix the tests by doing a POST /clients request after running a POST /pools to there would be a client available for the rest of the test suite.
2. Leave this programmatic client creation intact, which would result in the list of client always having at least 1 entry. Although approach 1 is preferred, I reintroduced this logic due to time constraints.

Reply:
@peterpolman Create a ticket for adjusting the tests